hw/ip/otbn/dv/rig/rig/gens/untaken_branch.py

```diff
@@ -0,0 +1,62 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+import random
+from typing import Optional
+
+from ..model import Model
+from ..program import ProgInsn, Program
+
+from .branch_gen import BranchGen
+
+
+class UntakenBranch(BranchGen):
+    '''A snippet generator for branches that are not taken.'''
+
+    def cal_offset(self,
+                   min_addr: int,
+                   max_addr: int,
+                   program: Program,
+                   model: Model) -> Optional[int]:
+        # Return None if this is the last instruction in the current gap
+        # because we need to either jump or do an ECALL to avoid getting stuck.
+        if program.get_insn_space_at(model.pc) <= 1:
+            return None
```

Comment:
This isn't really anything to do with the offset, so I don't think it belongs in this function.

I think it would probably be cleaner to implement `gen` in `UntakenBranch` and put this there, followed by a call to `super().gen(...)`.

As a bonus, it means that *this* function can now return an `int`, rather than an `Optional[int]`, which makes the call site nicer.